src/targets/gpu/fuse_mlir.cpp

```diff
@@ -44,22 +44,20 @@ struct mlir_op
 {
     std::string name() const { return "gpu::mlir_op"; }
     operation op = make_op("convolution");
+    shape output_shape;
 
     template <class Self, class F>
     static auto reflect(Self& self, F f)
     {
-        return pack(f(self.op, "op"));
+        return pack(f(self.op, "op"), f(self.output_shape, "output_shape"));
     }
 
     shape compute_shape(std::vector<shape> inputs, const std::vector<module_ref>& mods) const
     {
-        check_shapes{inputs, *this}.packed_or_broadcasted();
-        if(mods.size() != 1)
-            MIGRAPHX_THROW("should have one submodule.");
-        if(inputs.size() < 2)
-            MIGRAPHX_THROW("should have at least two inputs.");
-        auto n = inputs.size();
-        return op.compute_shape({inputs[n - 2], inputs[n - 1]});
+        // Currently fused mlir fusion uses static shapes.
```

Comment:
How important is it calculate the shape from the scratch here given that we only support static shaped kernels ?

Reply:
> How important is it calculate the shape from the scratch here given that we only support static shaped kernels ?

The shape can change after `eliminate_contiguous`. And most likely we would want to propagate the layout if it changes.

Reply:
ok added shape inference.